.github/actions/release/action.yml

```diff
@@ -1,59 +1,189 @@
-name: "tag version"
-description: "Tag package"
+name: "release action"
+description: "Create git release tags, github releases, jira version and push release communication. "
 inputs:
+  github_ref_name:
+    description: "Github ref name"
+    required: true
+  module_name:
+    description: "The package module name"
+    required: true
   jira_token:
     description: "Token to authenticate to Jira"
     required: true
   jira_project_id:
     description: "Jira project id to create release"
     required: true
+  jira_webhook_url:
+    description: "Jira release webhook"
+    required: true
+
 runs:
   using: "composite"
   steps:
-    - name: Publish RPMS to Repositories
+    - name: Checkout sources
+      uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
+    - name: Get released versions for components
       run: |
-        NEW_VERSION=""
-        MAJOR_VERSION=$(echo $GITHUB_REF_NAME | grep -oP '([0-9]{2}\.[0-9]{2})')
-        echo "Major version: $MAJOR_VERSION"
-        HOTFIX=$(echo $GITHUB_REF_NAME | grep -oP '(hotfix|)')
-        echo "Hotfix: $HOTFIX"
-        BETA=$(echo $GITHUB_REF_NAME | grep -oP '(beta|)')
-        echo "Beta: $BETA"
-        RELEASE_ID=$(git log -1 --pretty=%B | grep -oP '(#[0-9]{4,}#)' | grep -oP '([0-9]+)')
-        echo "Release Id: $RELEASE_ID"
-
-        OLDV=$(git tag --sort=-v:refname --list "centreon-collect-$MAJOR_VERSION.*" | head -n 1)
-        echo "Old version: $OLDV"
+        set -eu
+        COMPONENTS_RELEASED=()
+        COMPONENTS_OSS=("centreon-awie" "centreon-dsm" "centreon-gorgone" "centreon-ha" "centreon-open-tickets" "centreon-web")
+        COMPONENTS_MODULES=("centreon-anomaly-detection" "centreon-autodiscovery" "centreon-bam" "centreon-cloud-business-extensions" "centreon-cloud-extensions" "centreon-it-edition-extensions" "centreon-license-manager" "centreon-map" "centreon-mbi" "centreon-pp-manager")
+        COMPONENTS_COLLECT=("centreon-collect")
+        CURRENT_STABLE_BRANCH_MAJOR_VERSION=""
+        declare -a NEW_STABLE_TAGS=()
+        declare -a PREVIOUS_STABLE_TAGS=()
+        SCOPE_VERSION="COLLECT"
+        MINOR_VERSION_FILE_PATH=".version"
+
+        # Get current stable branch name
+        # CURRENT_STABLE_BRANCH_MAJOR_VERSION=$(git rev-parse --abbrev-ref HEAD | cut -d '.' -f1,2)
+        CURRENT_STABLE_BRANCH_MAJOR_VERSION="24.04"
+        echo "Curent stable branch major version: $CURRENT_STABLE_BRANCH_MAJOR_VERSION"
+
+        # Get previous version tags for components
+        for component in ${COMPONENTS_COLLECT[@]}; do
```

Comment:
Didn't understand. As there's only centreon-collect in this array set in line 32

Reply:
indeed, this loop could be skipped, I left it like this to be aligned with the other repositories, and also thinking of a new "component" becoming part of collect (what do you think @kduret )